Backport upstream patch for https://crbug.com/1015927

Change "-z defs" to "-Wl,-z,defs"

This is a linker option so make it more clearly understood by
passing it through the more standard "-Wl" syntax.
Unlike, "-Wl,..",  "-z defs" needs to be processed by compiler driver.

diff --git a/BUILD b/BUILD
index f43b8fd..335410c 100644
--- a/tensorflow/lite/BUILD
+++ b/tensorflow/lit/e/BUILD
@@ -429,7 +429,7 @@ tflite_cc_shared_object(
         ],
         "//tensorflow:windows": [],
         "//conditions:default": [
-            "-z defs",
+            "-Wl,-z,defs",
             "-Wl,--version-script,$(location //tensorflow/lite:tflite_version_script.lds)",
         ],
     }),
